lib/Command/ExpirePassword.php

```diff
@@ -106,54 +130,96 @@ protected function configure() {
 	 * @throws \OCP\PreConditionNotMetException
 	 */
 	protected function execute(InputInterface $input, OutputInterface $output) {
-		$uid = $input->getArgument('uid');
+		$groups = $input->getOption('group');
+		$allUsers = $input->getOption('all');
+		$uids = $input->getOption('uid');
 
-		/** @var $user \OCP\IUser */
-		$user = $this->userManager->get($uid);
-
-		if ($user === null) {
-			$output->writeln("<error>Unknown user: $uid</error>");
-			return self::EX_NOUSER;
+		$expireDate = new \DateTime();
+		$oldDate = new \DateTime();
+		$timeZone = new \DateTimeZone('UTC');
+		$inputExpireDate = $input->getArgument('expiredate');
+
+		if ($allUsers !== false) {
+			$this->userManager->callForAllUsers(function (IUser $user) use ($expireDate, $oldDate, $timeZone, $inputExpireDate, $output) {
+				$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+				$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+			});
+		} elseif (\count($groups) >= 1) {
+			$users = [];
+
+			foreach ($groups as $group) {
+				if ($this->groupManager->groupExists($group) === true) {
+					foreach ($this->groupManager->findUsersInGroup($group) as $user) {
+						if (!\in_array($user->getUID(), $users)) {
+							$calculatedExpireDate = $this->setPasswordExpiry($expireDate, $oldDate, $timeZone, $inputExpireDate, $user);
+							$output->writeln('The password for ' . $user->getUID() . ' is set to expire on ' . $calculatedExpireDate->format('Y-m-d H:i:s T') . '.');
+							$users[] = $user->getUID();
+						}
+					}
+				} else {
+					$output->writeln("<error>Group name $group doesn't exist</error>");
```

Comment:
Maybe changing the message for "Ignoring missing group $group" sounds more like a warning instead of an error. I'm not sure if it's possible to change the "error" tag to a "warning" tag.

Reply:
Didn't find a "warning" tag. These are the default ones available https://symfony.com/doc/current/console/coloring.html#using-color-styles. We can customise the colour. Like setting foreground and background colours.